freshmaker/handlers/koji/rebuild_images_on_rpm_advisory_change.py

```diff
@@ -451,6 +452,29 @@ def _find_images_to_rebuild(self, errata_id, skip_nvrs=None):
             "Going to find all the container images to rebuild as " "result of %r update.",
             affected_nvrs,
         )
+
+        adv_is_compliance_priority = getattr(self.event.advisory, "is_compliance_priority", False)  
+        # We should only use a restriction for external repos for advisories that are not critical
+        # or important, even if they have the 'compliance-priority' label
+        restrict_to_ext_repos = None
+        if adv_is_compliance_priority and self.event.advisory.security_impact not in ["critical", "important"]:
+            db_event = Event.get_or_create_from_event(db.session, self.event)
+            try:
+                restrict_to_ext_repos = self._lookup_external_repos()
+            except Exception as e:
+                msg = f"Unable to fetch external repos: {str(e)}"
+                self.log_error(msg)
+                db_event.transition(EventState.SKIPPED, msg)
```

Comment:
When error happens, we should update the event to `FAILED` state with the error message in `state_reason`.

Reply:
I see! I changed it to transition the event to 'failed' in case we get an HTTP error.

I had a look in `handle`, and I saw that there was a block that transitioned the state of the event to 'skipped' in case we returned an empty list from `_find_images_to_rebuild` [(link)](https://github.com/FernandesMF/freshmaker/blob/rebuild-from-predefined-repos/freshmaker/handlers/koji/rebuild_images_on_rpm_advisory_change.py#L102-L110).

Thus I added a new condition to that 'if' clause to not do that, in case the event state is 'failed'. That way, `handle` will not transition the event from 'failed' to skipped after an error happens. Is that okay?